### PR TITLE
Incorporate API Review feedback

### PR DESCRIPTION
**Pull request by bricelam**
Part of #24743

### CODE REVIEW

File: src/EFCore.InMemory/Infrastructure/InMemoryDbContextOptionsBuilder.cs [hunk 1]
```diff
@@ -46,14 +46,14 @@ DbContextOptionsBuilder IInMemoryDbContextOptionsBuilderInfrastructure.OptionsBu
...
+        /// <param name="nullChecksEnabled"> If <see langword="true" />, then nullability check is enforced. </param>
         /// <returns> The same builder instance so that multiple calls can be chained. </returns>
-        public virtual InMemoryDbContextOptionsBuilder EnableNullabilityCheck(bool nullabilityCheckEnabled = true)
+        public virtual InMemoryDbContextOptionsBuilder EnableNullChecks(bool nullChecksEnabled = true)
         {
             var extension = OptionsBuilder.Options.FindExtension<InMemoryOptionsExtension>()
                 ?? new InMemoryOptionsExtension();
 
-            extension = extension.WithNullabilityCheckEnabled(nullabilityCheckEnabled);
+            extension = extension.WithNullabilityCheckEnabled(nullChecksEnabled);
```
Comment (smitpatel) [inline src/EFCore.InMemory/Infrastructure/InMemoryDbContextOptionsBuilder.cs:56]: With method name?

File: test/EFCore.InMemory.FunctionalTests/NullabilityCheckInMemoryTest.cs [hunk 2]
```diff
@@ -31,7 +31,7 @@ public void IsRequired_for_property_throws_while_inserting_null_value()
 
                         var optionsBuilder = new DbContextOptionsBuilder()
                             .UseModel(modelBuilder.FinalizeModel())
-                            .UseInMemoryDatabase(nameof(NullabilityCheckInMemoryTest), b => b.EnableNullabilityCheck())
+                            .UseInMemoryDatabase(nameof(NullabilityCheckInMemoryTest), b => b.EnableNullChecks())
```
Comment (smitpatel) [inline test/EFCore.InMemory.FunctionalTests/NullabilityCheckInMemoryTest.cs:34]: may be change nameof arg?

File: src/EFCore.InMemory/Infrastructure/InMemoryDbContextOptionsBuilder.cs (hunk 1 above)
Comment (bricelam) [reply src/EFCore.InMemory/Infrastructure/InMemoryDbContextOptionsBuilder.cs:56]: Meh, it’s internal. (I actually did change it, but reverted before sending the PR.)

**Review by ajcvickers — APPROVED** [review]

File: src/EFCore.InMemory/Infrastructure/InMemoryDbContextOptionsBuilder.cs (hunk 1 above)
Comment (smitpatel) [reply src/EFCore.InMemory/Infrastructure/InMemoryDbContextOptionsBuilder.cs:56]: Well in that case, I am fine with anything.

**Pull request merged**